File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-13/automation/2026-07-13-control-evidence-approval-record-validation-tool.py
"""Synthetic AC-27 validation helper for p03-full-001822; never connects to a system."""
from __future__ import annotations

ORGANIZATION = "Cobalt Harbor Systems"
WORKING_PAPER_CONTEXT = {"stage": "intake-and-closure", "evidence_card": "EV-184-23", "review_lane": "PRIV-47", "verification_pass": 8, "working_paper_sequence": 1822, "spine_ids": ['org.cobalt-harbor', 'evidence.ev-184-19', 'rel.ac27.evidence-184', 'date.2026-07-13', 'term.no-kio-claim'], "rationale": 'The bounded intake and AC-27 closure record were checked against the same synthetic evidence chain.', "observation": 'Approval separation records preparer and reviewer roles for the synthetic exception disposition.'}
EXPECTED = {"control_id": "AC-27", "case_id": "CASE-260713-184", "risk_id": "RISK-031", "evidence_set": 184, "reviewed": 48, "timely_attestations": 47, "residual_risk": "Medium"}

def validate(record: dict[str, object]) -> list[str]:
    return [key for key, value in EXPECTED.items() if record.get(key) != value]

if __name__ == "__main__":
    print("Synthetic validation helper; provide records through an external safe harness.")

File: workspace/people/p03-security-grc-analyst/home/compliance/control-evidence/2026-07-13/automation/2026-07-13-control-evidence-approval-record-validation-tool.rs
// Synthetic AC-27 validation helper for p03-full-004830; no network or file access.
// stage=intake-and-closure; card=EV-184-07; lane=PRIV-31; pass=2; working_paper=4830; spine=org.cobalt-harbor, evidence.set-184, rel.ac27.evidence-184, date.2026-07-13, term.no-kio-claim
// The bounded intake and AC-27 closure record were checked against the same synthetic evidence chain. Approval separation records preparer and reviewer roles for the synthetic exception disposition. Evidence set 184 contains 24 synthetic items: 12 access-review attestations, 6 approval records, 4 checksum manifests, and 2 exception records.
const ORGANIZATION: &str = "Cobalt Harbor Systems";
pub fn valid(control_id: &str, case_id: &str, risk_id: &str, evidence_set: u32, reviewed: u32, timely: u32, residual: &str) -> bool {
    let _organization = ORGANIZATION;
    control_id == "AC-27" && case_id == "CASE-260713-184" && risk_id == "RISK-031" && evidence_set == 184 && reviewed == 48 && timely == 47 && residual == "Medium"
}
